### PR TITLE
react redux 7.1 (hooks)

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Updates react-redux to 7.1 which enables hooks
- [x] Ported over device page to use hooks
- [x] Hooks usage requires us changing how we mock out things in stories. Before we patched how connect worked but now we just make a fake store
- [x] Example mock for common things. In order for this to work with the new style and old style we entirely merge the old PropProvider keys and the real store shape

### CODE REVIEW

File: shared/.storybook/webpack.config.js [hunk 1]
```diff
@@ -34,7 +34,10 @@ module.exports = ({config, mode}) => {
       __STORYSHOT__: false,
       'process.platform': JSON.stringify('darwin'),
     }),
-    new webpack.NormalModuleReplacementPlugin(/^react-redux$/, __dirname + '/../__mocks__/react-redux.tsx'),
+    new webpack.NormalModuleReplacementPlugin(
```
Comment (chrisnojima) [inline shared/.storybook/webpack.config.js]: we just mock out Container.connect and not react-redux

File: shared/chat/conversation/input-area/normal/container.tsx [hunk 2]
```diff
@@ -214,8 +214,4 @@ const mergeProps = (stateProps, dispatchProps, ownProps: OwnProps): Props => ({
   unsentTextRefresh: !!stateProps.unsentText,
 })
 
-export default connect(
-  mapStateToProps,
-  mapDispatchToProps,
-  mergeProps
-)(Input)
+export default namedConnect(mapStateToProps, mapDispatchToProps, mergeProps, 'Input')(Input)
```
Comment (chrisnojima) [inline shared/chat/conversation/input-area/normal/container.tsx]: so this continues to be mocked the old way

File: shared/common-adapters/name-with-icon/index.stories.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ const innerClick = evt => {
 }
 
 const provider = Sb.createPropProviderWithCommon(
-  Sb.PropProviders.Avatar(['chrisnojima'], ['chrisnojima']),
+  Sb.PropProviders.Avatar(['both'], ['both']),
```
Comment (chrisnojima) [inline shared/common-adapters/name-with-icon/index.stories.tsx]: move over to a semantic special name instead of random usernames

File: shared/devices/device-page/container.tsx [hunk 4]
```diff
@@ -1,36 +1,23 @@
...
-})
-
-export default namedConnect(mapStateToProps, mapDispatchToProps, mergeProps, 'DevicePage')(DevicePage)
+// TODO(newRouter) after committing to new router:
+// remove action and code that sets state.devices.selectedDeviceID.
+// It's a bad pattern to have navigation distributed across react-navigation
+// and our store. device id is purely an argument to the screen, the store
+// doesn't care about it.
+export default Container.connect(
+  (state, ownProps: OwnProps) => ({id: Container.getRouteProps(ownProps, 'deviceID')}),
```
Comment (chrisnojima) [inline shared/devices/device-page/container.tsx]: this is an example of a component moving to hooks. we just pass through the id here. This whole thing could go away (onBack etc) if we changed the header hoc into a hook

File: shared/devices/device-page/index.stories.tsx [hunk 5]
```diff
@@ -1,49 +1,54 @@
...
-    .addDecorator(provider)
-    .add('Desktop', () => <DevicePage _type="desktop" />)
-    .add('Desktop no last used', () => <DevicePage _type="desktop" _lastUsed={false} />)
-    .add('Desktop current', () => <DevicePage _type="desktop" _current={true} />)
-    .add('Desktop Revoked', () => <DevicePage _type="desktop" _revoked={true} />)
-    .add('Mobile', () => <DevicePage _type="mobile" />)
-    .add('Mobile Revoked', () => <DevicePage _type="mobile" _revoked={true} />)
-    .add('Paper key', () => <DevicePage _type="backup" />)
-    .add('Paper key Revoked', () => <DevicePage _type="backup" _revoked={true} />)
+    .addDecorator((story: any) => <Sb.MockStore store={store}>{story()}</Sb.MockStore>)
```
Comment (chrisnojima) [inline shared/devices/device-page/index.stories.tsx]: this injects a navigation object since the component pulls the id out of the routeProps. We could add a layer of indirection here but we can do that later

File: shared/devices/device-page/index.tsx [hunk 6]
```diff
@@ -83,30 +84,36 @@ const Timeline = ({device}) => {
 }
 
 const DevicePage = (props: Props) => {
-  let metaOne
-  if (props.device.currentDevice) {
-    metaOne = 'Current device'
-  } else if (props.device.revokedAt) {
-    metaOne = <Kb.Meta title="revoked" style={styles.meta} backgroundColor={Styles.globalColors.red} />
-  }
+  const device = Container.useSelector(state => Constants.getDevice(state, props.id))
```
Comment (chrisnojima) [inline shared/devices/device-page/index.tsx]: ↪️  ↩️ 🎣 !!

File: shared/menubar/chat-container.desktop.tsx [hunk 7]
```diff
@@ -21,13 +21,15 @@ const mapDispatchToProps = dispatch => ({
...
 
 const mergeProps = (stateProps, dispatchProps, ownProps) => ({
-  convRows: stateProps.conversations
-    .slice(0, ownProps.convLimit ? ownProps.convLimit : stateProps.conversations.length)
-    .map(c => ({
-      conversationIDKey: c.conversationIDKey,
-      onSelectConversation: () => dispatchProps._onSelectConversation(c.conversationIDKey),
-      ...c,
-    })),
+  convRows: __STORYBOOK__
```
Comment (chrisnojima) [inline shared/menubar/chat-container.desktop.tsx]: i didn't fixup remote connect to participate in this so for now this makes it work

File: shared/stories/__tests__/__snapshots__/Storyshots.test.js.snap [hunk 8]
```diff
@@ -172937,7 +172937,7 @@ Array [
                     }
                   }
                 >
-                  chrisnojima
+                  both
```
Comment (chrisnojima) [inline shared/stories/__tests__/__snapshots__/Storyshots.test.js.snap]: stories are the same except i renamed this user

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/devices/device-page/container.tsx [hunk 9]
```diff
@@ -1,36 +1,23 @@
...
+  dispatch => ({
+    onBack: () => {
+      Container.isMobile && dispatch(RouteTreeGen.createNavigateUp())
+    },
+  }),
+  (stateProps, dispatchProps) => ({
+    id: stateProps.id,
+    onBack: dispatchProps.onBack,
+  })
+)(DevicePage)
```
Comment (songgao) [inline shared/devices/device-page/container.tsx]: It's using the hooks now; is there a reason this file still exists?

File: shared/profile/profile.stories.tsx [hunk 10]
```diff
@@ -69,7 +69,7 @@ const bioProps = {
   backgroundColorType: 'green' as BackgroundColorType,
   notAUser: false,
   onAddIdentity: Sb.action('onAddIdentity'),
-  onEditAvatar: Sb.action('onEditAvatarClick'),
+  onEditAvatar: null,
```
Comment (songgao) [inline shared/profile/profile.stories.tsx]: Is this change intended?

Comment (chrisnojima) [reply shared/profile/profile.stories.tsx]: yes

File: shared/devices/device-page/container.tsx (hunk 9 above)
Comment (chrisnojima) [reply shared/devices/device-page/container.tsx]: yes because of how the header hoc works it still needs a wrapping component
